Content.Shared/Storage/EntitySystems/StoreOnCollideSystem.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Content.Shared.Lock;
+using Content.Shared.Storage.Components;
+using Content.Shared.Whitelist;
+using Robust.Shared.Network;
+using Robust.Shared.Physics.Events;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Storage.EntitySystems;
+
+internal sealed class StoreOnCollideSystem : EntitySystem
+{
+    [Dependency] private readonly SharedEntityStorageSystem _storage = default!;
+    [Dependency] private readonly LockSystem _lock = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+    [Dependency] private readonly INetManager _netMan = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StoreOnCollideComponent, StartCollideEvent>(OnCollide);
+        SubscribeLocalEvent<StoreOnCollideComponent, StorageAfterOpenEvent>(AfterOpen);
+        // TODO: Add support to stop colliding after throw, wands will need a WandComp
+    }
+
+    // We use Collide instead of Projectile to support different types of interactions
+    private void OnCollide(Entity<StoreOnCollideComponent> ent, ref StartCollideEvent args)
+    {
+        TryStoreTarget(ent, args.OtherEntity);
+    }
+
+    private void AfterOpen(Entity<StoreOnCollideComponent> ent, ref StorageAfterOpenEvent args)
+    {
+        var comp = ent.Comp;
+
+        if (comp is { DisableWhenFirstOpened: true, Disabled: false })
+            comp.Disabled = true;
+    }
+
+    private void TryStoreTarget(Entity<StoreOnCollideComponent> ent, EntityUid target)
+    {
+        var storageEnt = ent.Owner;
+        var comp = ent.Comp;
+
+        if (_netMan.IsClient || _gameTiming.ApplyingState)
+            return;
+
+        if (ent.Comp.Disabled || storageEnt == target || Transform(target).Anchored || _storage.IsOpen(storageEnt) || _whitelist.IsWhitelistFail(comp.Whitelist, target))
```

Comment:
Wouldn't it be better for the locker to be open when shot and closing it when stopping? It is a little weird that is has to be closed to be able to store something, especially if this component would be reused elsewhere.

Reply:
The issue with that is people being able to leave the locker immediately while it's in transit. It needs to stay closed and arguably locked while flying or else it's kinda useless.

Reply:
Good point!